### ISSUE TITLE
if(!is_numeric($x)) should remove int and float types from $x

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#330]
Expected: No error
Observed: InvalidScalarArgument

```php
<?php

/**
 * @param int|string $x
 */
function test($x) : void {
    if (!is_numeric($x)) {
        echo strlen($x);  // Observed: InvalidScalarArgument - src/numeric.php:8:21 - Argument 1 of strlen expects string, int|string provided
    }
}
```

**Comment by muglug** [comment]
Thanks, great catch. I also added checks for `!is_scalar` in https://github.com/vimeo/psalm/commit/0c917f632daa0186a6abd19cf0270bec3997ddbf
